### PR TITLE
Fix pipeline setup under serverless

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## Proposed commit message

closes https://github.com/elastic/beats/issues/36965

This is a fairly simple fix, since the `PipelineLoader` is already backed by ES, so we just needed to expand the interface.

I have tested this under serverless, but I'm not too familiar with pipeline setup, so I would prefer that someone with more knowledge of filebeat setup also look at this. My understanding is that serverless has no API for plugins or ingest processors, so there's no need for us to perform this check under serverless.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: filebeat/fileset/pipelines.go [hunk 1]
```diff
@@ -65,16 +66,17 @@ func (reg *ModuleRegistry) LoadPipelines(esClient PipelineLoader, overwrite bool
...
 			reg.log.Debugf("Required processors: %s", requiredProcessors)
-			if len(requiredProcessors) > 0 {
+			// APIs do not exist on serverless
+			if len(requiredProcessors) > 0 && !esClient.IsServerless() {
 				err := checkAvailableProcessors(esClient, requiredProcessors)
 				if err != nil {
-					return fmt.Errorf("error loading pipeline for fileset %s/%s: %v", module.config.Module, fileset.name, err)
+					return fmt.Errorf("error loading pipeline for fileset %s/%s: %w", module.config.Module, fileset.name, err)
 				}
 			}
```
Comment (pchila) [inline filebeat/fileset/pipelines.go:75]: Disclaimer: I am not an expert on filebeat so the question may be naibe or obvious :sweat: 

Wouldn't it be a bit cleaner to have 1 empty set of processors for serverless in line 67 and load the regular sets of required processors for regular ElasticSearch and modify the `checkAvailableProcessors` to perform the ES call only if we pass a non-empty set of processors ?

The effect is the same but the code should look a bit cleaner.

Comment (fearful-symmetry) [reply filebeat/fileset/pipelines.go:75]: Maybe? One of the reasons why I chose that was that it made the code fairly explicit, I didn't want to bury the serverless logic too much.

File: filebeat/fileset/pipelines.go [hunk 2]
```diff
@@ -65,16 +66,17 @@ func (reg *ModuleRegistry) LoadPipelines(esClient PipelineLoader, overwrite bool
 			// check that all the required Ingest Node plugins are available
 			requiredProcessors := fileset.GetRequiredProcessors()
 			reg.log.Debugf("Required processors: %s", requiredProcessors)
-			if len(requiredProcessors) > 0 {
+			// APIs do not exist on serverless
+			if len(requiredProcessors) > 0 && !esClient.IsServerless() {
```
Comment (cmacknz) [inline filebeat/fileset/pipelines.go:70]: Just to double check, it is only the API used by `checkAvailableProcessors()` that is unimplemented?

That is this one API call? 
https://github.com/elastic/beats/blob/be28eb445e150ebcd213ece1e206841bb497eb0a/filebeat/fileset/modules.go#L395

That is using https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-nodes-info.html#cluster-nodes-info-api-example-ingest

This is only checking that the set of ingest processors we want to use is supported by the node we connected to. If that is the case this seems fine as long as loading the actual ingest pipelines actually works.

Did we test loading ingest pipelines? Attempting to install a module that defines an ingest pipeline is an easy way to test this.

Comment (fearful-symmetry) [reply filebeat/fileset/pipelines.go:70]: Yah, it's just that one API call. And yes, there's tests for pipelines as part of https://github.com/elastic/elastic-agent/pull/3658

Comment (fearful-symmetry) [reply filebeat/fileset/pipelines.go:70]: The existing tests are actually how I figured out this was broken in the first place.

**Review by cmacknz — APPROVED** [review]
Thanks!

**Pull request merged**